src/app/components/TableViewSpeedDial/index.tsx

```diff
@@ -0,0 +1,76 @@
+import SpeedDial from '@mui/material/SpeedDial'
+import SpeedDialAction from '@mui/material/SpeedDialAction'
+import PivotTableChartIcon from '@mui/icons-material/PivotTableChart'
+import TableRowsOutlinedIcon from '@mui/icons-material/TableRowsOutlined'
+import TableRowsIcon from '@mui/icons-material/TableRows'
+import ViewListOutlinedIcon from '@mui/icons-material/ViewListOutlined'
+import ViewListIcon from '@mui/icons-material/ViewList'
+import { FC, useState } from 'react'
+import { COLORS } from '../../../styles/theme/colors'
+import { useTranslation } from 'react-i18next'
+
+export enum TableView {
+  Horizontal = 'horizontal',
+  Vertical = 'vertical',
+}
+
+export interface TableViewSpeedDialProps {
+  tableView: TableView
+  setTableView: (tableView: TableView) => void
+}
+
+export const TableViewSpeedDial: FC<TableViewSpeedDialProps> = ({ tableView, setTableView }) => {
+  const { t } = useTranslation()
+  const [open, setOpen] = useState(false)
+
+  return (
+    <SpeedDial
+      sx={{ position: 'absolute', top: 0, right: 0, marginTop: -3 }}
```

Comment:
On small screens with a mouse: it is impossible to switch to another layout:
| Default mui for comparison | Issue |
| --- | --- |
| ![mui](https://user-images.githubusercontent.com/3758846/227670827-3465e2fb-f029-4f64-b246-19ebc10d7dfe.gif) | ![our](https://user-images.githubusercontent.com/3758846/227671206-34116995-37d8-4d1e-bfea-5ebb4eb99b5c.gif)   |

Reply:
Shouldn't matter after https://github.com/oasisprotocol/explorer/pull/205#discussion_r1151267408